agent/src/main/java/org/openremote/agent/protocol/tradfri/TradfriConnection.java

```diff
@@ -0,0 +1,133 @@
+package org.openremote.agent.protocol.tradfri;
+
+import org.openremote.agent.protocol.tradfri.device.Device;
+import org.openremote.agent.protocol.tradfri.device.Gateway;
+import org.openremote.agent.protocol.tradfri.device.Light;
+import org.openremote.agent.protocol.tradfri.device.Plug;
+import org.openremote.agent.protocol.tradfri.util.Credentials;
+
+import org.openremote.agent.protocol.ProtocolExecutorService;
+import org.openremote.model.asset.agent.ConnectionStatus;
+import org.openremote.model.attribute.AttributeEvent;
+import org.openremote.model.syslog.SyslogCategory;
+
+import java.util.function.Consumer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import static org.openremote.model.syslog.SyslogCategory.PROTOCOL;
+
+public class TradfriConnection {
```

Comment:
Not totally clear what the difference between a connection and a gateway is...they look to be just the same thing

Reply:
The Gateway class is the class that represents the IKEA TRÅDFRI gateway. This class is part of the IKEA TRÅDFRI library and is not dependent on OpenRemote classes.
The TradfriConnection contains all the methods and logic used by the OpenRemote protocol to manage a connection with the IKEA TRÅDFRI gateway (like the Consumers for the ConnectionStatus class).